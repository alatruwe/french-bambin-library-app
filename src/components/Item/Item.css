.vertical-border {
  display: inline-block;
  font-size: calc(5vw + 16px);
  line-height: 0.9;
  margin: 0 0.2em;
  width: 1.5vw;
  border-left: 0.2em solid rgba(0, 0, 0, 0);
  background: linear-gradient(#fff, #fff) padding-box,
    repeating-linear-gradient(
        -65deg,
        #5461c8 12.5vw,
        #c724b1 0 25vw,
        #e4002b 0 37.5vw,
        #ff6900 0 50vw,
        #f6be00 0 62.5vw,
        #97d700 0 75vw,
        #00ab84 0 87.5vw,
        #00a3e0 0 100vw
      )
      border-box;
  background-size: 200vw 100vh;
  background-attachment: fixed;
}

.item {
  display: flex;
}
.vertical-border {
  display: flex;
  flex-direction: column;
}
.item-details {
  display: flex;
  flex-direction: column;
}
.item-title {
  font-size: 25px;
}
.item-text {
  line-height: 1.5;
  padding: 3% 0;
}

.btn {
  width: auto;
  background: transparent;
  border: none;
  font-size: 18px;
}
.btn:after {
  content: "";
  display: block;
  width: 0%;
  padding-top: 5px;
  border-bottom: 1px solid black;
  transition: 0.5s;
}

.btn:hover:after,
.btn:focus:after {
  width: 100%;
}
