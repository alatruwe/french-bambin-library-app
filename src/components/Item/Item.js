import React from "react";
import ItemDetails from "../ItemDetails/ItemDetails";

export default function Item(props) {
  return (
    <section>
      <ItemDetails
        title={props.title}
        image={props.image}
        description={props.description}
      />
    </section>
  );
}
