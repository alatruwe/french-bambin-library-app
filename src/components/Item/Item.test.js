import React from "react";
import ReactDOM from "react-dom";
import Item from "./Item";
import renderer from "react-test-renderer";

describe(`Item component`, () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Item />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer.create(<Item />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
