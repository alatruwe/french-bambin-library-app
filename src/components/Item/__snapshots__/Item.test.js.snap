// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component renders the UI as expected 1`] = `
<section>
  <div>
    <div>
      <h2 />
      <img
        height="250"
        width="250"
      />
      <p />
    </div>
    <button
      type="button"
    >
      Send an request
    </button>
  </div>
</section>
`;
